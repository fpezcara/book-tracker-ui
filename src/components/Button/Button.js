import React from "react";
import { useHistory } from "react-router-dom";

const Button = ({ title, className, value, onClickHandler }) => {
  const { push } = useHistory();
  const onClickButton = () => {
    push(value);
    onClickHandler && onClickHandler();
  };
  return (
    <button className={className} onClick={onClickButton}>
      {title}
    </button>
  );
};

export default Button;
