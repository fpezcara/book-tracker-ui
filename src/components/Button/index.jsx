import React from "react";
import { useNavigate } from "react-router";

import { StyledButton } from "../../styles/Button.style";

const Button = ({ title, className, value, onClickHandler, dataTestId }) => {
  const navigate = useNavigate();
  const onClickButton = () => {
    value && navigate(value);
    onClickHandler && onClickHandler();
  };

  return (
    <StyledButton
      className={className}
      onClick={onClickButton}
      data-testid={dataTestId}
    >
      {title}
    </StyledButton>
  );
};

export default Button;
