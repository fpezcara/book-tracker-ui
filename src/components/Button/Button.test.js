import { render } from "@testing-library/react";
import Button from "./";
import userEvent from "@testing-library/user-event";

const mockNavigate = jest.fn();

jest.mock("react-router", () => ({
  useNavigate: () => mockNavigate,
}));

describe("Button", () => {
  it("renders correctly with given props", () => {
    const { container, getByText } = render(
      <Button
        title="accept"
        className="button accept"
        onClickHandler={jest.fn()}
        dataTestId="test-button"
      />,
    );

    expect(container).toMatchSnapshot();
    expect(getByText(/accept/i)).toBeVisible();
    expect(container.querySelector(".button.accept")).toBeInTheDocument();
  });

  it("calls onClickButton when button is clicked", async () => {
    const onClickButton = jest.fn();
    const { getByText } = render(
      <Button
        title="test"
        className="button accept"
        value="/testing"
        onClickHandler={onClickButton}
        dataTestId="test-button"
      />,
    );

    await userEvent.click(getByText(/test/i));

    expect(onClickButton).toHaveBeenCalledTimes(1);
  });

  describe("Navigation", () => {
    it("does not navigate if no value is provided", async () => {
      const { getByText } = render(
        <Button
          title="test"
          className="button accept"
          onClickHandler={jest.fn()}
          dataTestId="test-button"
        />,
      );

      await userEvent.click(getByText(/test/i));
      expect(mockNavigate).not.toHaveBeenCalled();
    });

    it("does navigates if value is provided", async () => {
      const { getByText } = render(
        <Button
          title="test"
          className="button accept"
          value="/testing"
          onClickHandler={jest.fn()}
          dataTestId="test-button"
        />,
      );

      await userEvent.click(getByText(/test/i));
      expect(mockNavigate).toHaveBeenCalledWith("/testing");
    });
  });
});
