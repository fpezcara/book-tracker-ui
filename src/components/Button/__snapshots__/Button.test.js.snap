// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Button renders correctly with given props 1`] = `
<div>
  <button
    class="sc-bRKDuR kQIrHC button accept"
    data-testid="test-button"
  >
    accept
  </button>
</div>
`;
